critiquebrainz/frontend/forms/review.py

```diff
@@ -16,7 +16,15 @@ def __call__(self, form, field):
         if form.state.data == "draft":
             return
         l = len(field.data) if field.data else 0
-        if l < self.min or self.max != -1 and l > self.max:
+        if (l < self.min or self.max != -1 and l > self.max) and (l != 0):
+            raise ValidationError(self.message)
+
+class TextAndRatingBothNotEmpty(object):
+    def __init__(self, message):
+        self.message = message
+
+    def __call__(self, form, field):
+        if not field.data and not form.rating.data:
```

Comment:
I did that - https://gist.github.com/ps2611/119b0d6e073ba680034f5a78789afe34
But somehow, it doesn't get added to form.errors and gets displayed like - https://gist.github.com/ps2611/1e3443a370865b6dfa1bfede07c950bc